packages/firestore/es5-pure-transformer.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from 'typescript';
+
+/**
+ * This TypeScript transformer marks ES5-downleveled classes as "pure functions", so that tools such
+ * as rollup and UglifyJS will drop them if they are unused in apps that depends on our libraries.
+ *
+ * It does so by looking at all synthetic "@class" comments generated by TypeScript down-leveling
+ * transforms and then appends a "#__PURE__" comment to each of them. The later is picked up by
```

Comment:
is there documentation for the `#__PURE__` attribute?

Reply:
Yup. Added comments inline.